src/lambda/gc02_check_group_access_configuration/app.py

```diff
@@ -0,0 +1,389 @@
+            marker = response.get("Marker")
+            if not marker:
+                break
+        return policies
+    except botocore.exceptions.ClientError as ex:
+        if "NoSuchEntity" in ex.response['Error']['Code']:
+            ex.response['Error']['Message'] = f"Unable to fetch policies for group '{group_name}'. No such entity found."
+        elif "InvalidInput" in ex.response['Error']['Code']:
+            ex.response['Error']['Message'] = f"Invalid group name '{group_name}' or marker '{marker}' input received."
+        else:
+            ex.response["Error"]["Message"] = "InternalError"
+            ex.response["Error"]["Code"] = "InternalError"
+        raise ex
+
+def lambda_handler(event, context):
+    """This function is the main entry point for Lambda.
+    Keyword arguments:
+    event -- the event variable given in the lambda handler
+    context -- the context variable given in the lambda handler
+    """
+    global AWS_CONFIG_CLIENT
+    global AWS_IAM_CLIENT
+    global AWS_ACCOUNT_ID
+    global AWS_S3_CLIENT
+    global EXECUTION_ROLE_NAME
+    global AUDIT_ACCOUNT_ID
+
+    evaluations = []
+    rule_parameters = {}
+
+    invoking_event = json.loads(event["invokingEvent"])
+    logger.info("Received event: %s", json.dumps(event, indent=2))
+
+    # parse parameters
+    AWS_ACCOUNT_ID = event["accountId"]
+
+    if "ruleParameters" in event:
+        rule_parameters = json.loads(event["ruleParameters"])
+
+    valid_rule_parameters = evaluate_parameters(rule_parameters)
+
+    if "ExecutionRoleName" in valid_rule_parameters:
+        EXECUTION_ROLE_NAME = valid_rule_parameters["ExecutionRoleName"]
+    else:
+        EXECUTION_ROLE_NAME = "AWSA-GCLambdaExecutionRole"
+
+    if "AuditAccountID" in valid_rule_parameters:
+        AUDIT_ACCOUNT_ID = valid_rule_parameters["AuditAccountID"]
+    else:
+        AUDIT_ACCOUNT_ID = ""
+
+    AWS_CONFIG_CLIENT = get_client("config", event)
+    AWS_IAM_CLIENT = get_client("iam", event)
+    AWS_S3_CLIENT = boto3.client("s3")
+    
+    # is this a scheduled invokation?
+    if is_scheduled_notification(invoking_event["messageType"]):
+        # yes, proceed
+        
+        admin_accounts_file_path = valid_rule_parameters.get("s3ObjectPath", "")
+        if check_s3_object_exists(admin_accounts_file_path) == False:
+            evaluations.append(
+                build_evaluation(
+                    event["accountId"],
+                    "NON_COMPLIANT",
+                    event,
+                    resource_type=DEFAULT_RESOURCE_TYPE,
+                    annotation="No AdminAccountsFilePath input provided.",
+                )
+            )
+        else:     
+            is_compliant = True
+            admin_accounts = read_s3_object(admin_accounts_file_path).splitlines()
+            
+            groups = fetch_groups()
+            for g in groups:
+                eval = check_group_policies(g.get("GroupName", ""), admin_accounts, event)
```

Comment:
Should the evaluations that are returned from `check_group_policies` be added to the `evaluations` list?

We have a function for batch-submitting evaluations when there is an arbitrary amount of them. 
```python
def submit_evaluations(
    aws_config_client, result_token: str, evaluations: list[dict], interval_between_calls: float = 0.25
):
    max_evaluations_per_call = 100
    while evaluations:
        batch_of_evaluations, evaluations = (
            evaluations[:max_evaluations_per_call],
            evaluations[max_evaluations_per_call:],
        )
        aws_config_client.put_evaluations(Evaluations=batch_of_evaluations, ResultToken=result_token)
        if evaluations:
            time.sleep(interval_between_calls)
```

Reply:
Yeah we can add them

Reply:
Not worried about hitting the max for this